webservices/phoenix_deployment.md

```diff
@@ -0,0 +1,192 @@
+# Deploying Phoenix/Elixir apps
+
+relevant libraries:
+- [erlang](https://elixir-lang.org/install.html#installing-erlang)
+- [elixir](https://elixir-lang.org/install.html)
+- [phoenix](https://hexdocs.pm/phoenix/installation.html)
+- [distillery](https://github.com/bitwalker/distillery)
+- [edeliver](https://github.com/edeliver/edeliver)
+
+deployment alternatives:
+- [heroku](https://hexdocs.pm/phoenix/heroku.html)
+- [gigalixir](http://gigalixir.readthedocs.io/en/latest/main.html#getting-started-guide)
+- [bootleg](https://github.com/labzero/bootleg)
+
+
+The most simple way of deploying a phoenix app is probably using [heroku](https://hexdocs.pm/phoenix/heroku.html) and is usually sufficient for simple apps with a limited lifespan, but heroku can start to get expensive pretty quickly to account for the convenience. The alternative is to deploy using a hosting service such as AWS, Google Cloud, Digital Ocean, etc. While less expensive, it does have the overhead of having to configure more of your environment to be able to serve your app.
+
+For the sake of this guide, we're going to make a few assumptions:
+- Two instances setup within one of these providers, one with ports 80 and 443 exposed. This would be considered your production environment.
+- Database setup (likely postgres) and accessible from your production instance.
+- Both running the same operating system/version. Likely ubuntu or similar.
+- Ssh access to both instances from your local machine.
+- Have a phoenix/elixir application you want to deploy.
+
+From here, the steps we need to accomplish are:
+1. Configure build and prod environments
+2. Install and configure [distillery](https://github.com/bitwalker/distillery)
+3. Install and configure [edeliver](https://github.com/edeliver/edeliver)
+4. Initiate a first deployment to verify the app is running.
+5. Make changes to the app and push up changes.
+
+## Configure build and prod environments
+For this guide, we're going to assume the setup of one build server and one production server. You can very easily expand this to include staging, dev, or any number of other environments you wish to deploy to and would only need to alter the deployment commands to reference that environment. You could also just use one server and have the same server act as both build and production.
+
+These steps will need to be repeated on each server unless specifically noted.
+- Ssh into server using non-root user setup on instance provisioning (ie ubuntu).
+- (Optional) install a version manager to manage erlang and elixir installations [asdf](https://github.com/asdf-vm/asdf) add elixir and erlang plugins.
+- Install [erlang](https://elixir-lang.org/install.html#installing-erlang) following the guide provided or via asdf.
+- Install [elixir](https://elixir-lang.org/install.html#unix-and-unix-like) following the guide provided or via asdf.
+- (Production) Create a folder `/home/ubuntu/app` and set the permissions with `cmod /home/ubuntu/app 755`
```

Comment:
`/home/<username>/app` depending on what you're using. But yeah, should default to ubuntu on AWS if using the right instance type